lib/util/utils.ts

```diff
@@ -1,8 +1,21 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
+import * as Long from "long";
 import * as log from "../log";
 import { generate_uuid, string_to_uuid } from "rhea-promise";
+import { isBuffer } from "util";
+
+/**
+ * A constant that indicates whether the environment is node.js or browser based.
+ */
+export const isNode = typeof navigator === "undefined" && typeof process !== "undefined";
```

Comment:
Have you see this pattern elsewhere?
I am wondering if this is the best way to decide `isNode`

Reply:
Yup. We have exported this function in `ms-rest-js` (runtime for Autorest generated TS libraries). It is a well tested method in context of Autorest generated libraries. Btw, found similar approaches on SO to detect whether one is in the browser or node.js environment.